packages/edit-navigation/src/hooks/use-navigation-editor.js

```diff
@@ -50,6 +50,20 @@ export default function useNavigationEditor() {
 		}
 	}, [ hasLoadedMenus ] );
 
+	useEffect( () => {
+		// Don't reset selected menu before we can check whether
+		// it is still available.
+		if ( ! hasLoadedMenus ) {
+			return;
+		}
+
+		const selectedMenu = menus?.find( ( { id } ) => id === selectedMenuId );
+
+		if ( ! menus?.length || ! selectedMenu ) {
+			setSelectedMenuId( 0 );
```

Comment:
I feel like `null` might better represent the absence of something. I know WordPress likes to count from `1`, but `0` still seems too much like a legit menu id.

Reply:
I tend to agree but I was basing it on

https://github.com/WordPress/gutenberg/blob/7fbe674c6619439e5b185a6fdae52f10812e9dd5/packages/edit-navigation/src/store/reducer.js#L92

Alternatively, we could create a new action `CLEAR_SELECTED_MENU_ID` which might make it more explicit about what the intention is.

Reply:
Yes, `404` originates from the Header component when it tried to fetch `menuName`. This code won't prevent that but will fix the broken screen issue.

We currently don't have a way to reset the selected menu ID if menus were deleted by any other means, e.g., WP CLI.

I agree that changing the default/cleared state to be `null` makes more sense now. I think @getdave mentioned this when reviewing original PR.

Reply:
I have an alternative PR that we also might want to consider. It's incoming...

Reply:
Alternative approach available https://github.com/WordPress/gutenberg/pull/32313